File: Backend/PCPB_backend/BusinessObjects/Models/OrderItem.cs
using System;
using System.Collections.Generic;

namespace DAOs;

public partial class OrderItem
{
    public int OrderItemId { get; set; }

    public int OrderId { get; set; }

    public int ProductId { get; set; }

    public int Quantity { get; set; }

    public decimal PriceAtOrder { get; set; }

    public decimal Subtotal { get; set; }

    public int ThirdPartyId { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public virtual Commission? Commission { get; set; }

    public virtual Order Order { get; set; } = null!;

    public virtual Product Product { get; set; } = null!;

    public virtual ThirdParty ThirdParty { get; set; } = null!;
}

File: app/src/main/java/com/example/pcbuilderguideapp/models/OrderItem.java
package com.example.pcbuilderguideapp.models;

import com.example.pcbuilderguideapp.utils.DateAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;
import java.util.Date;

public class OrderItem {
    @SerializedName("$id")
    private String id;

    @SerializedName("orderItemId")
    private int orderItemId;

    @SerializedName("orderId")
    private int orderId;

    @SerializedName("productId")
    private int productId;

    @SerializedName("quantity")
    private int quantity;

    @SerializedName("priceAtOrder")
    private double unitPrice;

    @SerializedName("totalPrice")
    private double totalPrice;

    @SerializedName("thirdPartyId")
    private int thirdPartyId;

    @SerializedName("productName")
    private String productName;

    @SerializedName("productImage")
    private String productImage;

    @SerializedName("thirdPartyName")
    private String thirdPartyName;

    @JsonAdapter(DateAdapter.class)
    @SerializedName("createdAt")
    private Date createdAt;

    @JsonAdapter(DateAdapter.class)
    @SerializedName("updatedAt")
    private Date updatedAt;

    // Getters
    public String getId() { return id; }
    public int getOrderItemId() { return orderItemId; }
    public int getOrderId() { return orderId; }
    public int getProductId() { return productId; }
    public int getQuantity() { return quantity; }
    public double getUnitPrice() { return unitPrice; }
    public double getTotalPrice() { return totalPrice; }
    public int getThirdPartyId() { return thirdPartyId; }
    public String getProductName() { return productName; }
    public String getProductImage() { return productImage; }
    public String getThirdPartyName() { return thirdPartyName; }
    public Date getCreatedAt() { return createdAt; }
    public Date getUpdatedAt() { return updatedAt; }
} 